File: JAVA/Pertemuan 01/src/id/ac/uph/pbo/pertemuan01/Mobil.java
package id.ac.uph.pbo.pertemuan01;

public class Mobil {

    // ── State (Atribut) ──────────────────────────────────────────────────────
    private String merek;
    private String warna;
    private int    kecepatan;          // km/jam
    private double jumlahBahanBakar;   // Liter

    // ── Constructor ──────────────────────────────────────────────────────────
    public Mobil(String merek, String warna, double bahanBakarAwal) {
        this.merek            = merek;
        this.warna            = warna;
        this.kecepatan        = 0;
        this.jumlahBahanBakar = bahanBakarAwal;
    }

    // ── Behavior (Method) ────────────────────────────────────────────────────
    public void jalan() {
        if (jumlahBahanBakar > 0) {
            kecepatan = 60;
            jumlahBahanBakar -= 5;   // konsumsi bahan bakar
            System.out.println("[" + merek + "] melaju " + kecepatan
                    + " km/j | BBM tersisa: " + jumlahBahanBakar + " L");
        } else {
            System.out.println("[" + merek + "] tidak bisa jalan – BBM habis!");
        }
    }

    public void berhenti() {
        kecepatan = 0;
        System.out.println("[" + merek + "] berhenti.");
    }

    public void klakson() {
        System.out.println("[" + merek + "] BEEP BEEP! 📯");
    }

    public void isiBahanBakar(double liter) {
        jumlahBahanBakar += liter;
        System.out.println("[" + merek + "] diisi BBM +" + liter
                + " L → total: " + jumlahBahanBakar + " L");
    }

    // ── Getter untuk menampilkan state ──────────────────────────────────────
    public void tampilkanState() {
        System.out.println("  Merek     : " + merek);
        System.out.println("  Warna     : " + warna);
        System.out.println("  Kecepatan : " + kecepatan + " km/j");
        System.out.println("  BBM       : " + jumlahBahanBakar + " L");
    }
}

File: PYTHON/Pertemuan 01/Mobil.py
import sys
sys.stdout.reconfigure(encoding='utf-8')
class Mobil:
    def __init__(self, merek: str, warna: str, bahan_bakar_awal: float):
        self.merek = merek
        self.warna = warna
        self.kecepatan = 0
        self.jumlah_bahan_bakar = bahan_bakar_awal

    def jalan(self):
        if self.jumlah_bahan_bakar > 0:
            self.kecepatan = 60
            self.jumlah_bahan_bakar -= 5
            print(f"[{self.merek}] melaju {self.kecepatan} km/j | BBM tersisa: {self.jumlah_bahan_bakar} L")
        else:
            print(f"[{self.merek}] tidak bisa jalan – BBM habis!")

    def berhenti(self):
        self.kecepatan = 0
        print(f"[{self.merek}] berhenti.")

    def klakson(self):
        print(f"[{self.merek}] BEEP BEEP! 📯")

    def isi_bahan_bakar(self, liter: float):
        self.jumlah_bahan_bakar += liter
        print(f"[{self.merek}] diisi BBM +{liter} L → total: {self.jumlah_bahan_bakar} L")

    def tampilkan_state(self):
        print(f"  Merek     : {self.merek}")
        print(f"  Warna     : {self.warna}")
        print(f"  Kecepatan : {self.kecepatan} km/j")
        print(f"  BBM       : {self.jumlah_bahan_bakar} L")

